ledfx/libraries/cache.py

```diff
@@ -0,0 +1,303 @@
+"""Image caching system for LedFx."""
+
+import hashlib
+import json
+import logging
+from datetime import datetime
+from pathlib import Path
+from typing import Optional
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ImageCache:
+    """
+    Manages cached images from remote URLs.
+
+    Cache Policy:
+    - No automatic expiration (cache and keep)
+    - No TTL-based refresh
+    - LRU eviction when size/count limits exceeded
+    - Explicit refresh/clear only via API
+    """
+
+    def __init__(
+        self,
+        config_dir: str,
+        max_size_mb: int = 500,
+        max_items: int = 500,
+    ):
+        """
+        Initialize image cache.
+
+        Args:
+            config_dir: LedFx configuration directory
+            max_size_mb: Maximum cache size in megabytes (default 500)
+            max_items: Maximum number of cached items (default 500)
+        """
+        self.cache_dir = Path(config_dir) / "cache" / "images"
```

Comment:
Should use pathlib or similar to join paths.

Reply:
yep